packaging/flavorgen/flavors/generators.go

```diff
@@ -396,6 +505,13 @@ func newHAProxyLoadBalancer() infrav1.HAProxyLoadBalancer {
 }
 
 func newKubeadmControlplane(replicas int, infraTemplate infrav1.VSphereMachineTemplate) controlplanev1.KubeadmControlPlane {
+	files := []bootstrapv1.File{
```

Comment:
This should only be here if using kube-vip. It is currently also being generated for the HAProxy flavor.